File: MiniDOMBase/DMD/Anagrafica/Tasks/StatiTaskLavorazione/StatiTasksDiLavorazione.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases
Imports minidom.Internals
Imports minidom.Anagrafica

Namespace Internals



    Public NotInheritable Class CStatiTasksDiLavorazioneClass
        Inherits CModulesClass(Of StatoTaskLavorazione)

        Friend Sub New()
            MyBase.New("modAnaStatoTaskLavorazione", GetType(StatoTaskLavorazioneCursor), -1)
        End Sub

        ''' <summary>
        ''' Restituisce il task attivo in base al nome
        ''' </summary>
        ''' <param name="name"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Function GetItemByName(ByVal name As String) As StatoTaskLavorazione
            Dim items As CCollection(Of StatoTaskLavorazione) = Me.LoadAll
            name = Strings.Trim(name)
            If (name = "") Then Return Nothing
            For Each stato As StatoTaskLavorazione In items
                If Strings.Compare(stato.Nome, name) = 0 Then Return stato
            Next
            Return Nothing
        End Function


    End Class

End Namespace

Partial Public Class Anagrafica

    Private Shared m_StatiTasksLavorazione As CStatiTasksDiLavorazioneClass = Nothing

    Public Shared ReadOnly Property StatiTasksLavorazione As CStatiTasksDiLavorazioneClass
        Get
            If (m_StatiTasksLavorazione Is Nothing) Then m_StatiTasksLavorazione = New CStatiTasksDiLavorazioneClass
            Return m_StatiTasksLavorazione
        End Get
    End Property

End Class

File: MiniDOMBase/DMD/Anagrafica/Tasks/StatiTaskLavorazione/StatiTasksDiLavorazione.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom.repositories;
using static minidom.Anagrafica;
using static minidom.Sistema;

namespace minidom
{
    namespace repositories
    {

        /// <summary>
        /// Repository di oggetti di tipo <see cref="StatoTaskLavorazione"/>
        /// </summary>
        public sealed class CStatiTasksDiLavorazioneClass 
            : CModulesClass<Anagrafica.StatoTaskLavorazione>
        {

            /// <summary>
            /// Costruttore
            /// </summary>
            public CStatiTasksDiLavorazioneClass() 
                : base("modAnaStatoTaskLavorazione", typeof(Anagrafica.StatoTaskLavorazioneCursor), -1)
            {
            }

            /// <summary>
            /// Restituisce il task attivo in base al nome
            /// </summary>
            /// <param name="name"></param>
            /// <returns></returns>
            /// <remarks></remarks>
            public Anagrafica.StatoTaskLavorazione GetItemByName(string name)
            {
                var items = LoadAll();
                name = DMD.Strings.Trim(name);
                if (string.IsNullOrEmpty(name))
                    return null;
                foreach (Anagrafica.StatoTaskLavorazione stato in items)
                {
                    if (DMD.Strings.Compare(stato.Nome, name, true) == 0)
                        return stato;
                }

                return null;
            }
        }
    }

    public partial class Anagrafica
    {
        private static CStatiTasksDiLavorazioneClass m_StatiTasksLavorazione = null;

        /// <summary>
        /// Repository di oggetti di tipo <see cref="StatoTaskLavorazione"/>
        /// </summary>
        public static CStatiTasksDiLavorazioneClass StatiTasksLavorazione
        {
            get
            {
                if (m_StatiTasksLavorazione is null)
                    m_StatiTasksLavorazione = new CStatiTasksDiLavorazioneClass();
                return m_StatiTasksLavorazione;
            }
        }
    }
}